main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -1389,16 +1389,27 @@ object Weeder2 {
                 case Pattern.Wild(loc) => Validation.success(WeededAst.RestrictableChoosePattern.Wild(loc))
                 case Pattern.Var(ident, loc) => Validation.success(WeededAst.RestrictableChoosePattern.Var(ident, loc))
                 case Pattern.Cst(Ast.Constant.Unit, loc) => Validation.success(WeededAst.RestrictableChoosePattern.Wild(loc))
-                case other => Validation.toHardFailure(UnsupportedRestrictedChoicePattern(isStar, other.loc))
+                case other =>
+                  val e = UnsupportedRestrictedChoicePattern(isStar, other.loc)
+                  val ident = Name.Ident(s"x${flix.genSym.freshId()}", other.loc.asSynthetic)
+                  Validation.toSoftFailure(WeededAst.RestrictableChoosePattern.Var(ident, other.loc.asSynthetic), e)
               }
-            case other => Validation.toHardFailure(UnsupportedRestrictedChoicePattern(isStar, other.loc))
+            case other =>
+              val e = UnsupportedRestrictedChoicePattern(isStar, other.loc)
+              val ident = Name.Ident(s"x${flix.genSym.freshId()}", other.loc.asSynthetic)
+              Validation.toSoftFailure(List(WeededAst.RestrictableChoosePattern.Var(ident, other.loc.asSynthetic)), e)
           }
           mapN(inner)(RestrictableChoosePattern.Tag(qname, _, loc))
-        case other => Validation.toHardFailure(UnsupportedRestrictedChoicePattern(isStar, other.loc))
+        case other =>
+          val e = UnsupportedRestrictedChoicePattern(isStar, other.loc)
+          val ident = Name.Ident(s"x${flix.genSym.freshId()}", other.loc.asSynthetic)
+          val patVar = WeededAst.RestrictableChoosePattern.Var(ident, other.loc)
```

Comment:
Why are we using `Var` here instead of a proper error node?

Reply:
Yeah, we should definitely have a proper error node.

Reply:
Tried adding an error node as a pattern but I can also try to do it as a var